bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -75,6 +75,21 @@ def run_plan(scope, plan_name, named_args = {})
       )
     end
 
+    # Check for `$BOLTDIR/.resource_types` and create reference based on file name
```

Comment:
I would like to do it in run_plan for two reasons
1. I can get the Boltdir path (we are already going to be looking up inventory for GH-1125 in run_plan)
2. We only need to register the types for plan execution, it seems wasteful to do it any time we need to do it in pal.

Reply:
On the flipside, in `run_plan` this will happen every time we run a plan, not just for the outer plan. It doesn't really take any time either way, though.

It just seems like it makes more sense to do it in PAL because it's part of setting up our execution environment rather than specifically part of executing a plan.

Reply:
Cool. I can look in to how to get a hold of the `boldir` path there. Honestly I did not really try too hard before, I just had been working on GH-1125 and knew I could easily get it in `run_plan`.

Reply:
Updated to do this in PAL.